src/main/java/view25d/view/TurtleView.java

```diff
@@ -100,20 +108,110 @@ public void manuallyRefreshReporterView(Context context)  throws LogoException {
 
     private void updateArrayList(Context context) throws LogoException {
         //turtleReporterValues.clear();
-        turtleReporterValues = new ArrayList<TurtleValue>();
+
+        ArrayList<TurtleValue> temp = new ArrayList<TurtleValue>();
+        ArrayList<LinkValue> temp2 = new ArrayList<LinkValue>();
+
         for (Agent a : myAgents.agents()) {
             Turtle turtle = (Turtle)a;
+            if (turtle.hidden()) {
+                continue;
+            }
             Color c = org.nlogo.api.Color.getColor(turtle.color());
             double val = (Double)reporter.report(context, new Object[]{turtle});
             double stemColor = getStemColor(context, turtle);
             TurtleValue tv = new TurtleValue( turtle.shape(), c, turtle.size(), turtle.xcor(), turtle.ycor(), val, stemColor);
-            turtleReporterValues.add(tv);
+            temp.add(tv);
+        }
+
+        // Get Set of Links associated with the Turtles
+        Set<Link> linkSet = getLinkSetFromTurtleSet(myAgents);
+
+        // Store the data needed for link visualization
+        for (Link link : linkSet) {
+
+            Turtle end1 = (Turtle)link.end1();
+            Turtle end2 = (Turtle)link.end2();
+
+            Color c = org.nlogo.api.Color.getColor(link.color());
+            double zcor1 = (Double)reporter.report(context, new Object[]{end1});
+            double zcor2 = (Double)reporter.report(context, new Object[]{end2});
+            
+            LinkValue lv = new LinkValue(link.shape(), c, link.lineThickness(),
+                                         end1.xcor(), end1.ycor(), zcor1,
+                                         end2.xcor(), end2.ycor(), zcor2);
+            temp2.add(lv);
         }
+
+        turtleReporterValues = temp;
+        linkValues = temp2;
+            
         if (viewOptions.usePColor()) {
             updatePColors();
+        }  
+    }
+
+
+    // Given an AgentSet of turtles, produce a Set of associated Links.
+    // Assumes check that AgentSet contains Turtles has already been made.
+    // Currently includes links for which only one end is in the AgentSet
+    public static  Set<Link>  getLinkSetFromTurtleSet(AgentSet turtleSet) {
+        
+        Link[] linkArray = getLinkArrayFromTurtleSet(turtleSet);
+
+        // Making it a set eliminates the duplicates
+        Set<Link> linkSet = new HashSet<>(Arrays.asList(linkArray));
+
+        return linkSet;
+        
+    }
+        
+    // Given an AgentSet of Turtles, produce an Array of associated links
+    // Assumes check that AgentSet contains Turtles has already been made.
+    // Each link will appear twice if both ends of a link are in the AgentSet
+    // 
+    public static Link[] getLinkArrayFromTurtleSet(AgentSet turtleSet) {
```

Comment:
The code here mostly looks fine, but is there a reason to return an array of `Link` objects instead of just creating and returning an `ArrayList<Link>`?  The `ArrayList` would simplify some of the code since you wouldn't have to manage the array size at all, and I don't see any array-specific features used.